### PR TITLE
Resolve scikit-learn conflict

### LINKED ISSUE #174: Some packages in Manifest are outdated
Some packages in Manifest.toml under examples are outdated, such as EnsembleKalmanProcesses. It leads to an error when running the Lorenz example:

```
LoadError: MethodError: no method matching ParameterDistribution(::Vector{Dict{String, Any}})
Closest candidates are:
  ParameterDistribution(::Union{Array{PDType}, PDType}, ::Union{Array{CType}, Array, CType}, ::Union{Array{ST}, ST}) where {PDType<:ParameterDistributionType, CType<:EnsembleKalmanProcesses.ParameterDistributions.ConstraintType, ST<:AbstractString} at ~/.julia/packages/EnsembleKalmanProcesses/OjLuD/src/ParameterDistributions.jl:167
Stacktrace:
 [1] top-level scope
   @ /central/home/zhaoyi/CalibrateEmulateSample.jl/examples/Lorenz/Lorenz_example.jl:123
 [2] include(fname::String)
   @ Base.MainInclude ./client.jl:451
 [3] top-level scope
   @ REPL[2]:1
in expression starting at /central/home/zhaoyi/CalibrateEmulateSample.jl/examples/Lorenz/Lorenz_example.jl:108
```

### LINKED ISSUE #176: SciKitLearn python backend update breaks GH testing
Old working version saw conda install 
`scikit-learn 1.1.1` and `scipy 1.8.1`
Now conda installs 
`scikit-learn 1.1.2` and `scipy 1.9.0`
I suspect the syntax for `pyimport` has changed

### PR DESCRIPTION
**Pull request by tsj5**
# PULL REQUEST

## Purpose and Content
This PR fixes the problems encountered by @szy21 in #175. The purpose of that PR was to update the Manifest for the examples (Project/Manifest for CES itself is unchanged). This PR uses @szy21 's updates and in addition pins the versions of scipy and scikit-learn installed through Conda.jl to avoid an incompatibility in libstdc++ (documented in #176).

## Benefits and Risks

**Benefits**: Resolves #176. @szy21's original PR resolves #174.

**Risks**: Introduces maintenance requirement of having to remember we've pinned the versions of some dependencies going forward; since these are python dependencies this information isn't in Project.toml, so we might forget about this in the future when we want/need to update scikit-learn. This is a very minor risk, as the functionality from sk-learn that we currently use is mature.

## Linked Issues
- Closes #174, #176 

## PR Checklist
- [X] This PR has a corresponding issue OR is linked to an SDI.
- [X] I have followed CliMA's codebase [contribution](https://clima.github.io/ClimateMachine.jl/latest/Contributing/) and [style](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) guidelines OR N/A.
- [X] I have followed CliMA's [documentation policy](https://github.com/CliMA/policies/wiki/Documentation-Policy).
- [X] I have checked all issues and PRs and I certify that this PR does not duplicate an open PR.
- [X] I linted my code on my local machine prior to submission: N/A.
- [X] Unit tests: N/A.
- [X] Code used in an integration test.
- [X] All tests ran successfully on my local machine.
- [X] All classes, modules, and function contain docstrings: N/A.
- [X] Documentation has been added/updated: N/A.

### CODE REVIEW

File: examples/Emulator/GaussianProcess/Manifest.toml [hunk 1]
```diff
@@ -106,22 +106,22 @@ uuid = "83423d85-b0ee-5818-9007-b63ccbeb887a"
 version = "1.16.1+1"
 
 [[deps.CalibrateEmulateSample]]
-deps = ["AbstractMCMC", "AdvancedMH", "Conda", "Distributions", "DocStringExtensions", "EnsembleKalmanProcesses", "GaussianProcesses", "LinearAlgebra", "MCMCChains", "Printf", "PyCall", "Random", "ScikitLearn", "Statistics", "StatsBase"]
-path = "../../.."
+deps = ["AbstractMCMC", "AdvancedMH", "Conda", "Distributions", "DocStringExtensions", "EnsembleKalmanProcesses", "GaussianProcesses", "LinearAlgebra", "MCMCChains", "Pkg", "Printf", "PyCall", "Random", "ScikitLearn", "Statistics", "StatsBase"]
+git-tree-sha1 = "5a93353fdc751a2116529e907316cc51275b77b8"
```
Comment (charleskawczynski) [inline examples/Emulator/GaussianProcess/Manifest.toml:110]: You probably don’t want to make these changes to CES.

Comment (tsj5) [reply examples/Emulator/GaussianProcess/Manifest.toml]: Thanks for the catch! Reverted in https://github.com/CliMA/CalibrateEmulateSample.jl/pull/177/commits/731abcd7527f843c62a9c74f5a69035e18a2801f.

**Comment by tsj5** [comment]
bors try

**Comment by tsj5** [comment]
Merging at request of @odunbar

**Review by odunbar — APPROVED** [review]

**Comment by odunbar** [comment]
I think Caltech HPC was down for maintenance yesterday which perhaps is why bors timed out
rerunning to check

**Comment by odunbar** [comment]
bors try

**Comment by odunbar** [comment]
bors try

**Comment by tsj5** [comment]
Test on linux failed due to known issue #169. Merging this PR under the expectation that this intermittent failure will be fixed in a future PR addressing that issue.

> ## try
> Build failed:
> 
> * [test-linux](https://github.com/CliMA/CalibrateEmulateSample.jl/runs/8155476051?check_suite_focus=true)

**Pull request merged**